### PR TITLE
Fix dataspace warnings in task logs

### PR DESCRIPTION
**Pull request by fviale**
Warnings or errors when transferring files using dataspaces used to appear in task logs but these logs disappeared (regression).
This commit fixes the issue.

### CODE REVIEW

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 1]
```diff
@@ -98,23 +99,25 @@
...
+
     /**
      * Mainly for testing purposes
      */
     TaskProActiveDataspaces() {
 
     }
 
-    public TaskProActiveDataspaces(TaskId taskId, NamingService namingService, boolean isRunAsUser) throws Exception {
+    public TaskProActiveDataspaces(TaskId taskId, NamingService namingService, boolean isRunAsUser,
```
Comment (mcompastie) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java:114]: To limit the impact of this PR, keeping the old constructor along side w/ the new one requiring the TaskLogger could have been an option.

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 2]
```diff
@@ -98,23 +99,25 @@
...
+            TaskLogger taskLogger) throws Exception {
         this.taskId = taskId;
         this.namingService = namingService;
         this.runAsUser = isRunAsUser;
         this.linuxOS = OperatingSystem.getOperatingSystem() == OperatingSystem.unix;
+        this.taskLogger = taskLogger;
```
Comment (mcompastie) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java:120]: The taskLogger attribute could have manipulated in a smoother way if it was implementing Optionnal<>

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 3]
```diff
@@ -375,12 +378,12 @@ private void logDataspacesStatus(String message, DataspacesStatusLevel level) {
...
         final boolean hasEol = message.endsWith(eol);
 
-        if (level == DataspacesStatusLevel.ERROR) {
-            this.clientLogs.append("[DATASPACES-ERROR] ").append(message).append(hasEol ? "" : eol);
-        } else if (level == DataspacesStatusLevel.WARNING) {
-            this.clientLogs.append("[DATASPACES-WARNING] ").append(message).append(hasEol ? "" : eol);
-        } else if (level == DataspacesStatusLevel.INFO) {
-            this.clientLogs.append("[DATASPACES-INFO] ").append(message).append(hasEol ? "" : eol);
+        if (taskLogger != null && level == DataspacesStatusLevel.ERROR) {
+            taskLogger.getErrorSink().println("[DATASPACES-ERROR] " + message + (hasEol ? "" : eol));
```
Comment (mcompastie) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java:382]: println() comes with an EOL characters (= CR/LF/CRLF). Therefor, I bet you wanted to use print(), and not println()

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 4]
```diff
@@ -849,55 +847,50 @@ public Boolean call() throws FileSystemException {
...
+                case TransferToUserSpace:
+                    if (USER != null) {
+                        toBeThrown = copyScratchDataToOutput(USER, "USER", os, selector, results);
                         break;
-                }
-
-                results.clear();
+                    }
+                case none:
+                    break;
```
Comment (mcompastie) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java:884]: Facultative

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java (hunk 3 above)
Comment (fviale) [reply scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java:382]: you are right

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java (hunk 1 above)
Comment (fviale) [reply scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java:114]: ok, I will keep the previous constructor

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java (hunk 4 above)
Comment (fviale) [reply scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java:894]: ok

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java (hunk 2 above)
Comment (fviale) [reply scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java:124]: I am not a fan of Optional (to say the least)

**Pull request merged**